lib/linked_list.rb

```diff
@@ -120,22 +199,46 @@ def has_cycle
     # Time Complexity:  
     # Space Complexity
     def get_first
-      raise NotImplementedError
+      if @head
+        return @head.data
+      else
+        return nil
+      end
     end
 
     # method that inserts a given value as a new last node in the linked list
     # Time Complexity:  
     # Space Complexity
     def add_last(value)
-      raise NotImplementedError
+    
+      new_node = Node.new(value, nil)
+      current = @head
+      if @head == nil
+        @head = new_node
+        return
+      else 
+        while current != nil
+          if current.next == nil
+            current.next = new_node
+            return "inserted"

```

Comment:
You probably don't need to return inserted, maybe return `true` or just the value that was inserted.